pkg/kubectl/cmd/new.go

```diff
@@ -0,0 +1,224 @@
+)
+
+func NewCmdNew(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "new TYPE [--name NAME] [--labels key1=val,key2=val]",
+		Short:   "Create a new object of the given type by filling in a template",
+		Long:    new_long,
+		Example: new_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunNew(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. One of: 'yaml' or 'json'.")
+	cmd.Flags().String("output-version", "", "Output the formatted object with the given version (default api-version).")
+	cmd.PersistentFlags().String("name", "", "The name of the new object")
+	cmd.PersistentFlags().String("labels", "", "A comma-separated list of labels in key=value form to apply to the created object")
+	cmd.Flags().Bool("dry-run", false, "If true, only print the example object, without editing or creating it.")
+
+	cmdutil.AddValidateFlags(cmd)
+
+	return cmd
+}
+
+func initExample(cmdNamespace string, resource string, f *cmdutil.Factory, cmd *cobra.Command) (runtime.Object, error) {
+	mapper, _ := f.Object()
+	version, kind, err := mapper.VersionAndKindForResource(resource)
+	if err != nil {
+		return nil, err
+	}
+
+	exBuilder := f.ExampleBuilder()
+	obj, _, err := exBuilder.NewExample(version, kind)
+	if err != nil {
+		return nil, err
+	}
+
+	mapping, err := mapper.RESTMapping(kind, version)
+	if err != nil {
+		return nil, err
+	}
+
+	err = mapping.SetAPIVersion(obj, version)
+	if err != nil {
+		return nil, err
+	}
+	err = mapping.SetKind(obj, kind)
+	if err != nil {
+		return nil, err
+	}
+	err = mapping.SetNamespace(obj, cmdNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	nameFlag := cmdutil.GetFlagString(cmd, "name")
+	if nameFlag != "" {
+		err = mapping.SetName(obj, nameFlag)
+	} else {
+		err = mapping.SetName(obj, " ")
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	labelsFlag := cmdutil.GetFlagString(cmd, "labels")
+	if labelsFlag != "" {
+		parsedLabels, err := kubectl.ParseLabels(labelsFlag)
+		if err != nil {
+			return nil, err
+		}
+
+		err = mapping.SetLabels(obj, parsedLabels)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return obj, nil
+}
+
+func RunNew(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		fmt.Fprint(out, "You must specify the type of resource to create. ", valid_resources)
+		return cmdutil.UsageError(cmd, "Required resource not specified.")
+	}
+	targetResource := args[0]
+	cmdNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	obj, err := initExample(cmdNamespace, targetResource, f, cmd)
+	if err != nil {
+		return err
+	}
+
+	schema, err := f.Validator(cmdutil.GetFlagBool(cmd, "validate"), cmdutil.GetFlagString(cmd, "schema-cache-dir"))
+	if err != nil {
+		return err
+	}
+
+	// if we are using dry run, just output directly
+	if cmdutil.GetFlagBool(cmd, "dry-run") {
```

Comment:
I think I like this as the default (maybe only) action and then allow piping to other commands rather than have an editor appear by default.  We have very few interactive commands.